compiler/testData/diagnostics/tests/nullabilityAndSmartCasts/kt1778.fir.kt

```diff
@@ -7,8 +7,8 @@ import checkSubtype
 fun main(args : Array<String>) {
     val x = checkSubtype<Any>(args[0])
     if(x is java.lang.CharSequence) {
-        if ("a" == x) x.<!UNRESOLVED_REFERENCE!>length<!> else x.length() // OK
-        if ("a" == x || "b" == x) x.<!UNRESOLVED_REFERENCE!>length<!> else x.length() // <– THEN ERROR
-        if ("a" == x && "a" == x) x.<!UNRESOLVED_REFERENCE!>length<!> else x.length() // <– ELSE ERROR
+        if (<!EQUALITY_NOT_APPLICABLE!>"a" == x<!>) x.<!UNRESOLVED_REFERENCE!>length<!> else x.length() // OK
```

Comment:
This is similar to https://github.com/JetBrains/kotlin/pull/4281#discussion_r607421688. BTW, I was surprised to learn `kotlin.String` is not a subtype of `java.lang.CharSequence`.